src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -701,6 +722,67 @@ public IEnumerable<string> GetCertificateKeysFromLocalState(TransactionOperation
             }
         }
 
+        public unsafe void PutUniqueItem(TransactionOperationContext context, string type, BlittableJsonReaderObject cmd, long index)
+        {
+            var items = context.Transaction.InnerTransaction.OpenTable(UniqueItemsSchema, Items);
+            cmd.TryGet(nameof(PutUniqueValueCommand.Key), out string key);
+            cmd.TryGet(nameof(PutUniqueValueCommand.Index), out long cmdIndex);
+            var dbKey = key.ToLowerInvariant();
+            using (Slice.From(context.Allocator, dbKey, out Slice keySlice))
+            using (items.Allocate(out TableValueBuilder tvb))
+            {
+                cmd.TryGet(nameof(PutUniqueValueCommand.Value), out BlittableJsonReaderObject value);
+                value = context.ReadObject(value, nameof(PutUniqueValueCommand.Value));
+
+                tvb.Add(keySlice.Content.Ptr, keySlice.Size);
+                tvb.Add(index);
+                tvb.Add(value.BasePointer, value.Size);
+
+                if (items.ReadByKey(keySlice, out var reader))
+                {
+                    var itemIndex = *(long*)reader.Read((int)UniqueItems.Index, out var _);
+                    if (cmdIndex == itemIndex)
+                    {
+                        items.Update(reader.Id, tvb);
+                        return;
+                    }
+                    throw new ConcurrencyException($"The key '{key}' was already changed by an other command.");
+                }
+                items.Set(tvb);
+            }
+            context.Transaction.InnerTransaction.LowLevelTransaction.OnDispose += transaction =>
+            {
+                if (transaction is LowLevelTransaction llt && llt.Committed)
+                    TaskExecutor.Execute(_ =>
+                    {
+                        try
+                        {
+                            _rachisLogIndexNotifications.NotifyListenersAbout(index, null);
+                        }
+                        catch (Exception e)
```

Comment:
This exception can occur during notification. If we have  concurrency exception we throw it during the apply.